### PR TITLE
Initial AWS Refactor

### PR DESCRIPTION
**Pull request by teohrt**
- Segregated frontend from backend to facilitate future deployments
- Converted express server into a GraphQL lambda function handler
- Added infrastructure as code (this isn't finished yet)
- Added logic to maintain local development functionality

### CODE REVIEW

File: backend/graphql/apolloServer.js [hunk 1]
```diff
@@ -89,29 +90,40 @@ module.exports = (dbClient, twilioClient, logger) => {
...
   };
 
   return new ApolloServer({
     typeDefs,
     resolvers,
-    context: ({ req }) => authController.apolloServerContextInit(req),
+    playground: {
+      endpoint: '/dev/graphql',
+    },
+    // Auth implementation:
```
Comment (aaa12345678910) [inline backend/graphql/apolloServer.js:115]: why is this commented out

File: backend/graphql/apolloServer.js [hunk 2]
```diff
@@ -89,29 +90,40 @@ module.exports = (dbClient, twilioClient, logger) => {
...
+        // validator.validateAuthorization(context.user.admin, 'createMeetingItem');
+        mutationResolver.createMeetingItem(args),
+      updateMeetingItem: async (_parent, args, context) =>
+        // validator.validateAuthorization(context.user.admin, 'updateMeetingItem');
+        mutationResolver.updateMeetingItem(args),
       createSubscription: async (_parent, args) => mutationResolver.createSubscription(args),
     },
   };
 
   return new ApolloServer({
```
Comment (aaa12345678910) [inline backend/graphql/apolloServer.js:109]: please comment a link to where we can find documentation on Apollo Server

File: backend/graphql/resolvers/validators.js [hunk 3]
```diff
@@ -2,7 +2,7 @@
 // I'm disabling this error because we're using variable names from the schema
 // that use snake_case since they're referencing values in our DB that also uses snake_case
 
-const { UserInputError, ForbiddenError } = require('apollo-server-express');
+const { UserInputError, ForbiddenError } = require('apollo-server-lambda');
```
Comment (aaa12345678910) [inline backend/graphql/resolvers/validators.js:5]: does this need to be conditional on isLambda

File: backend/handler.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+const logger = require('./utilities/logger');
+const dbClient = require('./db/dbClient')(logger);
+const twilioClient = require('./twilio/twilioClient')(logger);
+
+const server = require('./graphql/apolloServer')(dbClient, twilioClient, logger);
+
+if (process.env.IS_LAMBDA) {
+  exports.graphqlHandler = server.createHandler({
+    cors: {
+      origin: '*',
```
Comment (aaa12345678910) [inline backend/handler.js:13]: does this allow any origins? is this secure?

File: backend/heroku-server.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+const express = require('express');
+
+const app = express();
```
Comment (aaa12345678910) [inline backend/heroku-server.js:3]: will this be ran when someone runs the frontend locally?

File: backend/serverless.yml [hunk 6]
```diff
@@ -0,0 +1,150 @@
+service: gov-agenda-notifier
```
Comment (aaa12345678910) [inline backend/serverless.yml:1]: @darpham please verify this file

File: backend/graphql/apolloServer.js (hunk 1 above)
Comment (teohrt) [reply backend/graphql/apolloServer.js:115]: Fair point. Auth requires refactoring now that we're in AWS land. I'll remove the comments.

File: backend/handler.js (hunk 4 above)
Comment (teohrt) [reply backend/handler.js:13]: Yep. It's not secure enough for production, but it's allowing me to test the lambda. I'll add a comment highlighting that.

File: backend/heroku-server.js (hunk 5 above)
Comment (teohrt) [reply backend/heroku-server.js:3]: Nope. Local frontend development uses `react-scripts start`.

This is the server Heroku uses to host the frontend.

File: Procfile [hunk 7]
```diff
@@ -1 +1 @@
-web: node server/app.js
+web: node backend/heroku-server.js.js
```
[the remark this answers is not in the document]
Comment (teohrt) [reply Procfile:1]: Good catch. With Heroku tied to master, I'm not able to actually test this file until it's in master. Gross, but we're getting rid of Heroku soon anyway.

File: .gitignore [hunk 8]
```diff
@@ -96,6 +96,7 @@ dist
 
 # Serverless directories
 .serverless/
+secrets.json
```
[the remark this answers is not in the document]
Comment (teohrt) [reply .gitignore:99]: Yeeeep. I'll do that.

File: backend/graphql/apolloServer.js [hunk 9]
```diff
@@ -1,15 +1,16 @@
 /* eslint-disable max-len */
-const { ApolloServer, gql } = require('apollo-server-express');
+const isLambda = process.env.IS_LAMBDA;
+const { ApolloServer, gql } = isLambda ? require('apollo-server-lambda') : require('apollo-server');
 
 const getMutationResolver = require('./resolvers/mutation');
 const getQueryResolver = require('./resolvers/query');
-const getValidator = require('./resolvers/validators');
-const authController = require('../controllers/authController')();
+// const getValidator = require('./resolvers/validators');
```
[the remark this answers is not in the document]
Comment (teohrt) [reply backend/graphql/apolloServer.js:7]: I commented them out when I realized auth needed to be refactored now that we're in AWS land. I'll remove the comments.

File: backend/handler.js (hunk 4 above)
Comment (aaa12345678910) [reply backend/handler.js:13]: Can you create a github issue and add it to the todo comment

**Review by aaa12345678910 — DISMISSED** [review]

**Review by aaa12345678910 — APPROVED** [review]

**Pull request merged**